packages/eslint-config-skuba/no-sync-in-promise-iterable.test.ts

```diff
@@ -0,0 +1,365 @@
+import * as test from 'node:test';
+
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+RuleTester.afterAll = test.after;
+RuleTester.describe = test.describe;
+RuleTester.it = test.it;
+RuleTester.itOnly = test.it.only;
+
+import rule from './no-sync-in-promise-iterable.js';
+
+const ruleTester = new RuleTester({
+  languageOptions: {
+    parserOptions: {
+      projectService: {
+        allowDefaultProject: ['*.ts*'],
+      },
+      tsconfigRootDir: __dirname,
+    },
+  },
+});
+
+const methods = ['all', 'allSettled', 'any', 'race'];
+
+ruleTester.run('no-sync-in-promise-iterable', rule, {
+  valid: methods.flatMap((method) => [
+    {
+      code: `Promise.${method}()`,
+    },
+    {
+      code: `Promise.${method}([])`,
+    },
+    {
+      code: `Promise.${method}([,,])`,
+    },
+    {
+      code: `Promise.${method}([1, 2, 3])`,
```

Comment:
Would it be simpler to just ban any non promise results from being added to these? I don't see any value in:

```ts
Promise.all([1, 2, 3]);
```

or

Even if it's allowed in JavaScript.